### PR TITLE
Not changing dimension size for expand when target size is -1

### PR DESCRIPTION
**Pull request by ruotianluo**
#2350 .
Previously set size to -1 will produce a no dimension tensor. I hope no one ever did this otherwise this will break their code.

### CODE REVIEW

**Comment by gchanan** [comment]
should THLongStorage_inferSizeN, THLongStorage_inferSize2 support this as well?

**Comment by ruotianluo** [comment]
For now, THLongStorage_inferSizex are only used when input is some tensor size which will not include -1; so I decided not to add.

File: torch/_tensor_docs.py [hunk 1]
```diff
@@ -1717,10 +1717,11 @@ def callable(a, b) -> number
 expand(*sizes) -> Tensor
 
 Returns a new view of the tensor with singleton dimensions expanded
-to a larger size.
+to a larger size. -1 means not changing the size on that dimension.
```
Comment (gchanan) [inline torch/_tensor_docs.py]: nit: it's ambiguous what -1 means here...the size of a single dimension or the value of *sizes.  How about something like:

"Passing -1 as the size for a dimension means not changing the size of that dimension.  For example, if y is a tensor of size (1,10) then calling y.expand(3,-1) will expand y to be of size (3,10).

Comment (ruotianluo) [inline torch/_tensor_docs.py]: In a new paragraph?

Comment (gchanan) [inline torch/_tensor_docs.py]: Whatever you think is best.

**Review by soumith — APPROVED** [review]

**Comment by soumith** [comment]
thank you, this is now merged into master!

**Pull request closed without merging**